lib/active_merchant/billing/gateways/cyber_source.rb

```diff
@@ -556,7 +556,7 @@ def add_line_item_data(xml, options)
       end
 
       def add_merchant_data(xml, options)
-        xml.tag! 'merchantID', @options[:login]
+        xml.tag! 'merchantID', options[:merchant_id] || @options[:login]
```

Comment:
`options` vs `@options` here is interesting. This makes sense, since it looks like `@options` is a hash for "gateway options" and `options` is a hash for "request options", so `merchant_id` is passed after the gateway was originally created.